### PR TITLE
feat(stepfunctions-tasks): add EKS Create/Delete Clusters to SFN-tasks

### PR DESCRIPTION
**Pull request by Sumeet-Badyal**
feat(stepfunctions-tasks): support for APIGW API: Invoke

Implementation

Update package @aws-cdk/aws-stepfunctions-tasks to include support for EKS Create Cluster and Delete Cluster
API as per documentation here:
https://docs.aws.amazon.com/step-functions/latest/dg/connect-eks.html
https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateCluster.html
https://docs.aws.amazon.com/eks/latest/APIReference/API_DeleteCluster.html

Includes support for the following Amazon EKS API calls:
CreateCluster
CreateCluster.sync
DeleteCluster
DeleteCluster.sync

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/delete-cluster.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: [
+          Stack.of(this).formatArn({
+            service: 'eks',
+            resource: 'cluster',
+            resourceName: this.props.name,
+          }),
+        ],
+        actions: ['eks:DeleteCluster'],
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/delete-cluster.ts:47]: Run a Job pattern will also need eks:DescribeCluster.

File: packages/@aws-cdk/aws-stepfunctions-tasks/test/eks/create-cluster.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import * as cdk from '@aws-cdk/core';
+import { EksCreateCluster, EncryptionConfig } from '../../lib/eks/create-cluster';
+
+describe('Create a Cluster', () => {
+
+  test('default settings', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const encryptoinConfiguration : EncryptionConfig = {
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/test/eks/create-cluster.test.ts:10]: Typo: should be 'encryption' not 'encryptoin'.

File: packages/@aws-cdk/aws-stepfunctions-tasks/test/eks/create-cluster.test.ts [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        ClientRequestToken: 'clientRequestToken',
+        EncryptionConfig: [encryptoinConfiguration],
+      },
+    });
+  });
+
+  test('createCluster.sync', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const encryptoinConfiguration : EncryptionConfig = {
```
Comment (russmackenzie) [inline packages/@aws-cdk/aws-stepfunctions-tasks/test/eks/create-cluster.test.ts:67]: Typo: should be 'encryption' not 'encryptoin'.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  readonly name: string;
+
+  /**
+   * The desired Kubernetes version for your cluster.
+   * @default - latest
+   */
+  readonly kubernetesVersion?: string;
+
+  /** EKS Role ARN to create a cluster */
+  readonly role: string;
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:18]: type should be `iam.IRole`

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+
+  /** VPC configuration used by the cluster control plane */
+  readonly resourcesVpcConfig: VpcConfigRequest;
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:21]: why not `ec2.IVpc` as it's used everywhere in the CDK (and within the stepfunctions module)

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/delete-cluster.ts [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      }),
+    ];
+  }
+
+  /**
+   * Provides the EKS delete cluster service integration task configuration
+   */
+  /**
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/delete-cluster.ts:55]:
```suggestion
  /**
   * Provides the EKS delete cluster service integration task configuration
   *
```

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+    this.integrationPattern = props.integrationPattern ?? sfn.IntegrationPattern.REQUEST_RESPONSE;
+
+    validatePatternSupported(this.integrationPattern, EksCreateCluster.SUPPORTED_INTEGRATION_PATTERNS);
+
+    this.taskPolicies = [
+      new iam.PolicyStatement({
+        resources: ['*'],
+        actions: ['eks:CreateCluster'],
+      }),
+      new iam.PolicyStatement({
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:76]: add a comment for a policy statement that has a `*` in it

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+  readonly role: string;
+
+  /** VPC configuration used by the cluster control plane */
+  readonly resourcesVpcConfig: VpcConfigRequest;
+
+  /**
+   * Enable or disable exporting the Kubernetes control plane logs for your cluster to CloudWatch Logs.
+   * @default - no logging configuration
+   */
+  readonly loggingConfiguration?: LoggingConfig;
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:27]: by convention for the most part the types for configurable properties are suffixed `Options` - i.e. `LoggingOptions`

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts [hunk 9]
```diff
@@ -0,0 +1,225 @@
...
+   * Metadata that you apply to the cluster to assist with categorization and organization
+   * @default - no tags
+   */
+  readonly tags?: { [key: string]: string };
+
+  /**
+   * The encryption configuration for the cluster.
+   * @default - No encryption config
+   */
+  readonly encryptionConfig?: EncryptionConfig[];
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:45]: is no encryption a good default for EKS? - i would think that encryption is enabled by default with an ability to turn it off.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts [hunk 10]
```diff
@@ -0,0 +1,225 @@
...
+            EndpointPrivateAccess: this.props.resourcesVpcConfig.endpointPrivateAccess,
+            PublicAccessCidrs: this.props.resourcesVpcConfig.publicAccessCidrs,
+          },
+          ClientRequestToken: this.props.clientRequestToken,
+          Tags: this.props.tags,
+          EncryptionConfig: this.props.encryptionConfig,
+        }),
+      };
+    }
+  }
```
Comment (shivlaks) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:135]: there's a fair amount of duplication here - let's simplify it down to one block.

**Review by shivlaks — CHANGES_REQUESTED** [review]
@Sumeet-Badyal - I took a bit of a first pass here, but did not go through all of it yet. I think we need to make some changes to the types first to get it looking more like other CDK construct libraries as well as start to reuse the types within the CDK.

File: packages/@aws-cdk/aws-stepfunctions-tasks/README.md [hunk 11]
```diff
@@ -657,6 +660,41 @@ new tasks.EmrModifyInstanceGroupByName(stack, 'Task', {
...
+
+[Read more](https://docs.aws.amazon.com/step-functions/latest/dg/connect-eks.html) about the differences when using these service integrations.
+
+### EKS Create Cluster
+
+Creates and starts running a cluster.
+Corresponds to the [`createCluster`](https://docs.aws.amazon.com/eks/latest/APIReference/API_CreateCluster.html) API in EKS.
+
+```ts
+new tasks.EksCreateCluster(stack, 'Create a Cluster', {
```
[the remark this answers is not in the document]
Comment (Sumeet-Badyal) [reply packages/@aws-cdk/aws-stepfunctions-tasks/README.md]: Some of the individual parameters do not map 1:1 and some of the required parameters for ICluster are unnecessary when using Step Functions (clusterEndpoint, clusterCertificateAuthorityData, etc)

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts (hunk 9 above)
Comment (Sumeet-Badyal) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:67]: From what I've seen in usage in the Console and docs here (https://aws.amazon.com/blogs/containers/using-eks-encryption-provider-support-for-defense-in-depth/) under How does it work?, one has to enable encryption.

Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/eks/create-cluster.ts:67]: okay, let's point to the documentation about how it needs to be enabled.
feels strange to be able to configure something that cannot be enabled here.

**Review by shivlaks — CHANGES_REQUESTED** [review]
took another pass briefly - feedback so far

**Pull request closed without merging**